server/storage/mvcc/kvstore_txn.go

```diff
@@ -114,15 +118,23 @@ func (tr *storeTxnRead) rangeKeys(ctx context.Context, key, end []byte, curRev i
 				zap.Int("len-values", len(vs)),
 			)
 		}
+		totalBytes += int64(len(vs[0]))
+		if ro.MaxBytes > 0 && totalBytes > ro.MaxBytes {
+			break
+		}
 		if err := kvs[i].Unmarshal(vs[0]); err != nil {
 			tr.s.lg.Fatal(
 				"failed to unmarshal mvccpb.KeyValue",
 				zap.Error(err),
 			)
 		}
+		count += 1
+		if ro.MaxBytes > 0 && totalBytes == ro.MaxBytes {
```

Comment:
This is premature optimization. I expect that cost of calculating this on every loop is larger than saving we would get. 

Don't know what values of maxBytes you expect users to select, but let's say user sets 1MB. What's the probability that byte sizes align exactly to 1 MB?

Reply:
Do you want to combine this to the condition at line 122 (ie if `ro.MaxBytes > 0 && totalBytes >= ro.MaxBytes`)

Reply:
This is not just optimization but also the correctness. We still need this KV pair if the total size equals to `maxBytes`

Reply:
Please see comment above about how similar for loop can be made simpler.

Reply:
Done